pkg/controller/jobframework/interface.go

```diff
@@ -116,6 +117,13 @@ type ComposableJob interface {
 	ForEach(f func(obj runtime.Object))
 }
 
+// JobWithCustomWorkloadConditions interface should be implemented by generic jobs,
+// when custom workload conditions should be updated after ensure that the workload exists.
+type JobWithCustomWorkloadConditions interface {
+	// CustomWorkloadConditions return custom workload conditions and status changed or not.
+	CustomWorkloadConditions(wl *kueue.Workload) ([]metav1.Condition, bool)
```

Comment:
Can we instead put it somewhere inside the implementation of `FindMatchingWorkloads` for the pod integration? We are already removing excess pods and doing patches to the Workload object there.

Reply:
I think that `FindMatchingWorkloads` already doing so much. The function name is `Find...` that is shouldn't update workload. So, if we can move this logic to another place, that's better in my opinion.